### DISCUSSION TITLE
How to build facet search based on taxonomy

### DISCUSSION THREAD

**Discussion by ns8482e** [OrchardCMS/OrchardCore#10514] [Q&A]
Is there any guidance or document available using Lucene query to build facet search.

**Comment by Skrypt** [comment]
I don't think there is any Faceted search implemented yet. Only normal search.
We would need to implement this : https://github.com/NightOwl888/BoboBrowse.Net
This is maintained by the same guy that does Lucene.NET

**Reply by ns8482e** [reply]
Orchard1.x has facet search correct?

**Comment by giannik** [comment]
yes it did.
https://www.youtube.com/watch?v=7v5qSR4g7E0&t=2825s

With elasticsearch it will be easier.

**Reply by Skrypt** [reply]
As you can see in this demo @sebastienros is using a custom API controller. So he basically creates a facet search based on current Lucene indexed elements. That's what does BoboBrowse I guess. I agree that ElasticSearch will make this easier as it is already implemented.

What we should do is implement this in the Lucene Queries : https://www.elastic.co/guide/en/app-search/current/facets.html

**Reply by ns8482e** [reply]
Do we have `GetBits` in OrchardCore?

**Reply by Skrypt** [reply]
I don't think they implemented this part at all. The idea was to use the ElasticSearch Queries and to implement QueryProviders instead.

**Reply by Skrypt** [reply]
https://www.nuget.org/packages/Lucene.Net.Facet
https://lucene.apache.org/core/4_8_0/facet/index.html

And more specifically for Taxonomies : https://github.com/apache/lucenenet/tree/master/src/Lucene.Net.Facet/Taxonomy

**Reply by ns8482e** [reply]
For time being - I am using `TaxonomyIndex` to get count  - works for basic scenario 
Except - I can't rebuild index for existing content items - I am republishing all items.

**Comment by MikeKry** [comment]
just fyi: there is a way in current impl. by using HighFreqTerms. But there probably are still problems with rebuilding index for existing content items. as mentioned by @ns8482e 

(i was also trying this: https://github.com/aspcodenet/MultiFacetLuceneNet/pull/4 but it was not reliable)

ex. 
![image](https://user-images.githubusercontent.com/52829889/160401666-dcce4d98-8b4f-4bae-bd97-26d44f244a57.png)


or using simplefacets based on usecase... => https://github.com/apache/lucenenet/blob/master/src/Lucene.Net.Demo/Facet/SimpleFacetsExample.cs
